src/NHibernate/Util/EnumerableExtensions.cs

```diff
@@ -6,6 +6,16 @@ namespace NHibernate.Util
 {
 	public static class EnumerableExtensions
 	{
+		public static HashSet<T> ToHashSet<T>(this IEnumerable<T> collection)
```

Comment:
>Also, if you scroll a bit down you will see ForEach() extension method.

We are discouraged of using these extensions, and the plan to eliminate their usage.